Add member repository

jaekkang opened the pull request:
Member Rpository 작성 및 해당 부분 테스트 추가하였습니다.

jaekkang committed:
Init: Create member repository

멤버 데이터를 저장하기 위한 저장소를 만들었습니다.
기본적인 영속성 추가를 위한
save()와 단건 데이터 검색과 전체 조회 메서드를 추가했습니다.

jaekkang committed:
Refactor: Change MemberRepository type

MemberRepository의 형식을 JPA Repository를 상속받는 형식으로 수정하였습니다.

jaekkang committed:
Feat: Add method for findByName

name을 기준으로 검색을 하는 메서드를 추가했습니다.

jaekkang committed:
Merge branch 'master' into AddMemberRepository

jaekkang committed:
Refactor: Change logic in create method

생성 메서드를 Builder 사용하는 방식으로 수정하였습니다.

jaekkang committed:
Style: Change method name

잘못 작성한 메서드의 이름을 변경하였습니다.
findByName -> findByusername

jaekkang committed:
Test: Add MemberRepository Test

작성한 Member Repository Test 추가했습니다.

간단히
- 빌더통해 생성 확인
- 저장한 데이터 불러오기
- 잘못된 색인시 에러 나는지

jaekkang committed:
Feat: Add hasher

Hasher 부분이 사용되지 않은 것을 확인해서 추가했습니다.

kshshkim commented on src/main/java/beforespring/socialfeed/member/domain/MemberRepository.java:
findByUsername의 반환값을 Optional로 감싸는 것이 좋아보입니다.

kshshkim commented on src/test/java/beforespring/socialfeed/member/domain/MemberRepositoryTest.java:
data jpa repository의 경우에는 추가로 선언한 메소드만 테스트하면 될 것 같습니다.

kshshkim replied:
테스트 케이스를 지울 필요는 없어요

jaekkang replied:
어라 테스트 케이스를 지울 필요는 없다는 메세지만 이제봄.. 뭐야 깃헙

kshshkim replied:
PasswordHasherImpl이 누락된 것 같습니다.

jaekkang committed:
Feat: Remove validate

validator가 잘못된 곳에서 사용되고 있어서 삭제했습니다.

jaekkang committed:
Feat: Add optional for empty data

Repository에서 서치시 데이터가 없을 경우를 위해 Optional로 감쌋습니다.

jaekkang committed:
Test: remove default method

빌더를 통한 객체 생성 테스트를 제외하고 jpaRepository에서 자동 생성되는 메서드의 테스트는 삭제했습니다.

kshshkim commented on src/main/java/beforespring/socialfeed/member/domain/Member.java:
PasswordHasher를 메서드 파라미터 부분에서도 지워주셔야합니다. 
그리고 updatePassword에는 Password가 유효한지 검사하는 로직이 포함이 되어야합니다.

추가로 메서드에 무슨 역할을 하는 메서드인지 주석 추가해주시면 좋을 것 같습니다.

jaekkang committed:
Style: Change method name

- validatePassword를 의미적으로 좀 더 맞는
verifyPassword로 변경되었습니다.

- 메서드에 대한 설명도 추가되었습니다.

jaekkang committed:
Test: Change test

Test문이 줄어 빌더 생성관련 밖에 남지 않아 MemberTest로 변경되었습니다.
